File: leetcode/11.js
/**
 * @param {number[]} height
 * @return {number}
 */


 var maxArea = function(height) {
    let maxArea = 0;
    let area;
    let currentHeight;
    let left = 0;
    let right = height.length - 1;
    if(height.length === 0 || height.length === 1){
        return maxArea;
    }
    while(left < right){
        width = right - left;
        if(height[left] < height[right]){
            area = width * height[left];
            left++;
        }else{
            area = width * height[right];
            right--;
        }
        maxArea = area > maxArea ? area : maxArea;
    }
    return maxArea;
};

File: leetcode/11.py
def maxContainer(height):
    maxArea = 0
    i = 0
    j = len(height) - 1
    while(i < j):
        if height[i] < height[j]:
            smallestLine = height[i]
            area = smallestLine * (j - i)
            i += 1
        else:
            smallestLine = height[j]
            area = smallestLine * (j - i)
            j -= 1
        if area > maxArea:
            maxArea = area
    return maxArea


print(maxContainer([0.0]))
